performance/cinterop/src/main/kotlin-native/org/jetbrains/cinteropBenchmarks/structsBenchmark.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.structsBenchmarks
+import kotlinx.cinterop.*
+import kotlin.math.sqrt
+
+const val benchmarkSize = 10000
+
+actual fun structBenchmark() {
+    memScoped {
+        val containsFunction = staticCFunction<CPointer<ElementS>?, CPointer<ElementS>?, Int> { first, second ->
+            var result: Int;
+            if (first == null || second == null) {
+                result = 0
+            } else if (first.pointed.string.toKString().contains(second.pointed.string.toKString())) {
+                result = 1
+            }
+            else {
+                result = 0
+            }
+            result
+        }
+        val elementsList = mutableListOf<ElementS>()
+        // Fill list.
+        for (i in 1..benchmarkSize) {
+            val element = alloc<ElementS>()
+            element.floatValue = i + sqrt(i.toDouble()).toFloat()
+            element.integer = i.toLong()
+            itoa(i, element.string, 10)
+            element.contains = containsFunction
+
+            elementsList.add(element)
+        }
+        val summary = elementsList.map { multiplyElementS(it.readValue(), (0..10).random()) }
+                .reduce { acc, it -> sumElementSPtr(acc.ptr, it.ptr)!!.pointed.readValue() }
+        val intValue = summary.useContents { integer }
+        elementsList.last().contains!!(elementsList.last().ptr, elementsList.first().ptr)
```

Comment:
```
elementsList.last().apply { contains!!(this, elementsList.first().ptr) }
```

Reply:
`inline fun <T> T.apply(block: T.() -> Unit): T`
But my function return Int.

Reply:
Do you mean that using it with `T = Int` may cause redundant boxing?
`elementsList.last()` should at least be extracted to a variable.

Reply:
I ment that contains has return value type Int, so `block: T.() -> Int` by Unit is expected